packages/world/ts/actions/callFrom.ts

```diff
@@ -52,12 +52,17 @@ type SystemFunction = { systemId: Hex; systemFunctionSelector: Hex };
 // The function mapping is cached to avoid redundant retrievals for the same World function.
 export function callFrom<TChain extends Chain, TAccount extends Account>(
   params: CallFromParameters,
-): (client: WalletClient<Transport, TChain, TAccount>) => Pick<WalletActions<TChain, TAccount>, "writeContract"> {
+): (client: Client<Transport, TChain, TAccount>) => Pick<WalletActions<TChain, TAccount>, "writeContract"> {
   return (client) => ({
     // Applies to: `client.writeContract`, `getContract(client, ...).write`
     writeContract: async (writeArgs): Promise<WriteContractReturnType> => {
-      // Skip if the contract isn't the World.
-      if (writeArgs.address !== params.worldAddress) {
+      // Skip if the contract isn't the World or the function called should not be redirected through `callFrom`.
+      if (
+        writeArgs.address !== params.worldAddress ||
+        writeArgs.functionName === "call" ||
+        writeArgs.functionName === "callFrom" ||
+        writeArgs.functionName === "registerDelegationWithSignature"
```

Comment:
Is this `registerDelegationWithSignature` condition necessary? The `registerDelegationWithSignature` function should be called by a delegator, not a delegatee. This `callFrom` extension action is intended for the use of the delegatee's wallet client (after delegation), and the delegator's wallet client does not need to use this.

Reply:
`registerDelegationWithSignature` _is_ called by the delegatee, using a signed message by the delegator!